packages/bsky/src/api/app/bsky/graph/getSuggestedFollowsByActor.ts

```diff
@@ -0,0 +1,135 @@
+import { sql } from 'kysely'
+import { Server } from '../../../../lexicon'
+import AppContext from '../../../../context'
+import { InvalidRequestError } from '@atproto/xrpc-server'
+
+const MAX_RESULTS_LENGTH = 10
+
+export default function (server: Server, ctx: AppContext) {
+  server.app.bsky.graph.getSuggestedFollowsByActor({
+    auth: ctx.authVerifier,
+    handler: async ({ auth, params }) => {
+      const { actor } = params
+      const viewer = auth.credentials.did
+
+      const db = ctx.db.getReplica()
+      const actorService = ctx.services.actor(db)
+      const actorDid = await actorService.getActorDid(actor)
+
+      if (!actorDid) {
+        throw new InvalidRequestError('Actor not found')
+      }
+
+      const actorsViewerFollows = db.db
+        .selectFrom('follow')
+        .where('creator', '=', viewer)
+        .select('subjectDid')
+
+      /**
+       * 20 most liked accounts that aren't already followed by the viewer, ARE
+       * the viewer, or are the actor
+       */
+      const mostLikedAccounts = db.db
+        .selectFrom(
+          db.db
+            .selectFrom('like')
+            .where('creator', '=', actorDid)
+            .select(sql`split_part(subject, '/', 3)`.as('subjectDid'))
+            .as('likes'),
```

Comment:
can we put a limit of 1000 on this subquery? otherwise it'll end up iterating over every like (which will be pretty expensive for some accounts